CMakeLists.txt

```
@@ -545,12 +545,15 @@ if (LLAMA_VULKAN)
 endif()
 
 if (LLAMA_HIPBLAS)
-    if ($ENV{ROCM_PATH})
-        set(ROCM_PATH $ENV{ROCM_PATH})
-    else()
-        set(ROCM_PATH /opt/rocm)
+    if (NOT EXISTS $ENV{ROCM_PATH})
+        if (NOT EXISTS /opt/rocm)
+            set(ROCM_PATH /usr)
+        else()
+            set(ROCM_PATH /opt/rocm)
+        endif()
     endif()
     list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH})
+    list(APPEND CMAKE_PREFIX_PATH "$ENV{ROCM_PATH}/lib64/cmake")
```

Is the `ENV` correct here? Shouldn't it use the `ROCM_PATH` variable set previously?

```suggestion
    list(APPEND CMAKE_PREFIX_PATH "${ROCM_PATH}/lib64/cmake")
```

---

It should be without ENV but for some reason CMake fails to find some package configuration without it.
```
-- Detecting HIP compile features
-- Detecting HIP compile features - done
CMake Error at /usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  By not providing "FindAMDDeviceLibs.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "AMDDeviceLibs", but CMake did not find one.

  Could not find a package configuration file provided by "AMDDeviceLibs"
  with any of the following names:

    AMDDeviceLibsConfig.cmake
    amddevicelibs-config.cmake

  Add the installation prefix of "AMDDeviceLibs" to CMAKE_PREFIX_PATH or set
  "AMDDeviceLibs_DIR" to a directory containing one of the above files.  If
  "AMDDeviceLibs" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /opt/rocm_sdk_611/lib64/cmake/hip/hip-config-amd.cmake:66 (find_dependency)
  /opt/rocm_sdk_611/lib/cmake/hip/hip-config.cmake:149 (include)
  CMakeLists.txt:583 (find_package)


-- Configuring incomplete, errors occurred!
```

---

Found the issue, I forgot to set ROCM_PATH in case the ENV exists, I'll fix it right now.